src/ComponentHelper.js

```diff
@@ -1,25 +1,40 @@
 var deepEqual = require('deep-equal');
+const cameraWorldPosition = new THREE.Vector3();
+const cameraGroupConjugate = new THREE.Quaternion();
 
 module.exports.gatherComponentsData = function(el, schemaComponents) {
-  var elComponents = el.components;
   var compsData = {};
 
   for (var i in schemaComponents) {
     var element = schemaComponents[i];
 
     if (typeof element === 'string') {
-      if (elComponents.hasOwnProperty(element)) {
+      if (el.components.hasOwnProperty(element)) {
         var name = element;
-        var elComponent = elComponents[name];
-        compsData[name] = AFRAME.utils.clone(elComponent.data);
+        // In VR mode, aframe gives over control of the camera to three's WebVRManager. WebVRManager modifies
+        // the camera's matrixWorld directly, so we have to decompose the camera's position ourselves.
+        // However, the camera's world position incorporates the camera's rotation in a weird way, so we have to
+        // undo that rotation as well.
+        if (name === 'position' && el.components.hasOwnProperty('camera') && el.sceneEl.is('vr-mode')) {
+          el.components.camera.camera.getWorldPosition(cameraWorldPosition);
+          cameraGroupConjugate.copy(el.object3D.quaternion);
+          cameraGroupConjugate.conjugate();
+          cameraWorldPosition.sub(el.object3D.position);
+          cameraWorldPosition.applyQuaternion(cameraGroupConjugate);
+          cameraWorldPosition.add(el.object3D.position);
+          compsData[name] = AFRAME.utils.clone(cameraWorldPosition);
+        }
+        else {
+          compsData[name] = AFRAME.utils.clone(el.getAttribute(name));
+        }
       }
     } else {
       var childKey = NAF.utils.childSchemaToKey(element);
       var child = element.selector ? el.querySelector(element.selector) : el;
       if (child) {
-        var comp = child.components[element.component];
-        if (comp) {
-          var data = element.property ? comp.data[element.property] : comp.data;
+        if (child.components.hasOwnProperty(element.component)) {
+          var attributeData = child.getAttribute(element.component);
+          var data = element.property ? attributeData[element.property] : attributeData;
```

Comment:
This can just become
```
var data = child.getAttribute(element.component, element.property);
```
since `getAttribute` can handle getting sub properties for you when an argument is passed.

Reply:
Really? I don't see that documented anywhere.

Reply:
I thought you could too but apparently not: https://github.com/aframevr/aframe/blob/master/src/core/a-entity.js#L726

I tried in the console too and it didn't work.

Reply:
hmm, could have sworn this used to work...